benchmarks/distributed/lambda_runtimes/cpp_runtime/main.cpp

```diff
@@ -0,0 +1,217 @@
+    // convert to wchar
+    std::vector<wchar_t> vec;
+    auto len = strlen(task_root);
+    vec.resize(len + 1);
+    mbstowcs(&vec[0], task_root, len);
+    Py_SetPythonHome(&vec[0]);
+
+    // initialize interpreter
+    // Py_Initialize();
+    if (!Py_IsInitialized()) {
+        PyErr_Print();
+        r = -1;
+        return "Couldn't initialize interpreter!";
+    }
+    timings.PYTHON_READY_TS = timestamp();
+
+    // grab the payload
+    JsonValue json(payload);
+    auto v = json.View();
+    auto func_str = v.GetString("function");
+    auto pickled_func_str = decodeFunction(func_str, r);
+    if (r == -1) {
+        res = pickled_func_str;
+        return res.c_str();
+    }
+
+    /*auto py_func_str = PyUnicode_FromString(func_str.c_str());
+
+    // pass to lambda_handler
+    auto sys_path = PySys_GetObject("path");
+    PyList_Append(sys_path, PyUnicode_FromString(task_root));
+    auto lambda_module = PyImport_ImportModule("lambda_function");
+    if (!lambda_module) {
+        PyErr_Print();
+        return invocation_response::success("Couldn't open module lambda_function", "text/plain");
+    }
+    auto lambda_handler = PyObject_GetAttrString(lambda_module, "lambda_handler");
+    if (!lambda_handler) {
+        PyErr_Print();
+        return invocation_response::success("Couldn't get function lambda_handler", "text/plain");
+    }
+
+    // call the handler
+    auto py_func_res = PyObject_CallFunctionObjArgs(lambda_handler, py_func_str);
+
+    // save function results
+    auto func_res = PyLong_AsLong(py_func_res);
+    timings.FUNCTION_TS = timestamp();
+
+    // shut down the interpreter
+    //Py_Finalize();
+     */
+
+    // use tuplex instead of native Python functions
+    auto py_module = python::getMainModule();
+    auto py_func = python::deserializePickledFunction(py_module, pickled_func_str.c_str(),
+                                                      pickled_func_str.length());
+    tuplex::ExceptionCode ec;
+    auto py_func_res = python::callFunction(py_func, PyTuple_New(0), ec);
+    auto func_res = PyFloat_AsDouble(py_func_res);
+    timings.FUNCTION_TS = timestamp();
+
+    timings.setString(func_res);
+    return res.c_str();
+}
+
+static invocation_response my_handler(invocation_request const &req) {
+    try {
+        int r;
+        handler_worker(req.payload.c_str(), r);
+
+        if(r == -1) {
+            return invocation_response::success(res, "plain/text");
+        } else {
+            return invocation_response::success(res, "application/json");
+        }
+    } catch(const std::exception& e) {
+        return invocation_response::success(std::string("exception! ") + e.what(), "plain/text");
+    } catch(...) {
+        return invocation_response::success(std::string("unhandled exception!"), "plain/text");
+    }
+}
+// --- BASICALLY THE SAME AS awslambda/ -------------------------------------------
+int main() {
+    timings.MAIN_TS = timestamp();
```

Comment:
did we ever update to the more recent AWS C++ Lambda runtime? If not, maybe worth a thought.